builtin/credential/approle/validation.go

```diff
@@ -335,23 +334,98 @@ func (b *backend) secretIDAccessorLock(secretIDAccessor string) *sync.RWMutex {
 	return lock
 }
 
+// nonLockedSecretIDStorageEntry fetches the secret ID properties from physical
+// storage. The entry will be indexed based on the given HMACs of both role
+// name and the secret ID. This method will not acquire secret ID lock to fetch
+// the storage entry. Locks need to be acquired before calling this method.
+func (b *backend) nonLockedSecretIDStorageEntry(s logical.Storage, roleNameHMAC, secretIDHMAC string) (*secretIDStorageEntry, error) {
+	if secretIDHMAC == "" {
+		return nil, fmt.Errorf("missing secret ID HMAC")
+	}
+
+	if roleNameHMAC == "" {
+		return nil, fmt.Errorf("missing role name HMAC")
+	}
+
+	// Prepare the storage index at which the secret ID will be stored
+	entryIndex := fmt.Sprintf("secret_id/%s/%s", roleNameHMAC, secretIDHMAC)
+
+	entry, err := s.Get(entryIndex)
+	if err != nil {
+		return nil, err
+	}
+	if entry == nil {
+		return nil, nil
+	}
+
+	result := secretIDStorageEntry{}
+	if err := entry.DecodeJSON(&result); err != nil {
+		return nil, err
+	}
+
+	persistNeeded := false
+	if result.SecretIDNumUsesDeprecated != 0 {
```

Comment:
The net result of this is that after the first time this is run `SecretIDNumUsesDeprecated` will no longer have a value...which means if they're relying on that value now, they'll only see it (and the warning) once rather than it going away in a future release.

The best option I can think of is adding another if condition below that, if `result.SecretIDNumUses` is less than `SecretIDNumUsesDeprecated` (and the latter isn't zero), the current value of `result.SecretIDNumUses` gets copied to `SecretIDNumUsesDeprecated` and then persisted. That way the values will be in proper lockstep during lookups.

If this sounds horrid, maybe we should just break things -- as you pointed out, it only came out last release so nobody is likely hugely depending on this value right now.